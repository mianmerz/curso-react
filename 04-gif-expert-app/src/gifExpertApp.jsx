import { useState } from "react"
import { AddCategory } from "./components/AddCategory";
import { GifGrid } from "./components/GifGrid";

export const GifExpertApp = () => {

    const [categories, setCategories] = useState(['One Punch']);
    console.log("categories", categories);

    const onAddCategory = (category) => {
        console.log("new category", category);

        if (categories.includes(category)) return;

        setCategories(categories => [category, ...categories]);

        // setCategories(cat => [...cat, "valorant"]);
        // setCategories([...categories,"A"]);
    }



    return (
        <>
            {/* titulo */}
            <h1>Git Expert App </h1>

            {/* Input */}
            <AddCategory
                onNewCategory={onAddCategory}
            // setCategories={setCategories} 
            />

            {/* Listado de Gif */}
            {
                categories.map(category => (
                    <GifGrid key={category} category={category} />
                ))
            }
            {/* {categories.map(c => <li key={c}> {c} </li>)} */}

            {/* Gif Item */}
        </>
    )
}
