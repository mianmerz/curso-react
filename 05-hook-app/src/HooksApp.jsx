import { CounterApp } from "./01-useState/CounterApp"
import { CounterWithCustomHook } from "./01-useState/CounterWithCustomHook"
import { SimpleForm } from './02-useEffect/SimpleForm';
import { FormWithCustomHook } from './02-useEffect/FormWithCustomHook';
import { MultipleCustomHooks } from "./03-examples/MultipleCustomHooks";
import { FocusScreen } from "./04-useRef/FocusScreen";
import { Layout } from "./05-useLayoutEffect/Layout";
import { Memorize } from "./06-memo/Memorize";
import { MemoHook } from "./06-memo/MemoHook";
import { CallbackHook } from "./06-memo/CallbackHook";

export const HooksApp = () => {
    return (
        <>
            <div>HooksApp</div>

            {/* <CounterApp /> */}
            {/* <CounterWithCustomHook /> */}
            {/* <SimpleForm /> */}
            {/* <FormWithCustomHook /> */}
            {/* <MultipleCustomHooks /> */}
            {/* <FocusScreen /> */}
            {/* <Layout /> */}
            {/* <Memorize /> */}
            {/* <MemoHook /> */}
            <CallbackHook />
        </>
    )
}
