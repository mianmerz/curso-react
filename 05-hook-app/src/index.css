body{
    padding: 20px;
}

button{
    margin-right: 10px;
}

section, .section{
    margin-top: 40px;
    border: 1px solid #ccc;
    padding: 20px;
    border-radius: 5px;
}