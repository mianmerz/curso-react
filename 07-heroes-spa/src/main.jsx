import React from 'react'
import ReactDOM from 'react-dom/client'
import './styles.css'

ReactDOM.createRoot(document.getElementById('root')).render(
  <React.StrictMode>
    <h1>asd</h1>
  </React.StrictMode>,
)
