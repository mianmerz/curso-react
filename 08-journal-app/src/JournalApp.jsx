
export const JournalApp = () => {
  return (
    <div>Hola mundito</div>
  )
}
