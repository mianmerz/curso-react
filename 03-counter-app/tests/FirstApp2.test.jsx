/** @jest-environment jsdom */
import { render, screen } from "@testing-library/react";
import { FirstApp } from "../src/FirstApp";


describe('Pruebas en <FirstApp />', () => {
    const title = "Hola, soy Migue";
    const subtitle = "Soy un subtitulito";

    test('Debe de hacer Match con el Snapshot', () => {
        const { container } = render(<FirstApp title={title} />);
        expect(container).toMatchSnapshot();
    });


    test('Debe de mostrar el mensaje "Hola, soy Migue', () => {
        render(<FirstApp title={title} />);
        // screen.debug();
        expect(screen.getByText(title)).toBeTruthy();
    });


    test('Debe de mostrar el titulo en un H1', () => {
        render(<FirstApp title={title} />);
        expect(screen.getByRole("heading", { level: 1 }).innerHTML).toContain(title)
    });


    test('Debe de mostrar el subtitulo enviado por props', () => {
        render(<FirstApp title={title} subtitle={subtitle} />);

        expect(screen.getAllByText(subtitle).length).toBe(2);
    });
});