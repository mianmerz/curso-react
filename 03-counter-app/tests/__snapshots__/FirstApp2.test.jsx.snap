// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <FirstApp /> Debe de hacer Match con el Snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
     
    Hola, soy Migue
     
  </h1>
  <p />
  <p />
  <p>
    Miguel Merelli 
  </p>
</div>
`;
